southbound/mqtt/mqtt-client/src/main/java/org/eclipse/sensinact/gateway/southbound/mqtt/impl/MqttClientHandler.java

```diff
@@ -115,24 +124,39 @@ public void activate(final MqttClientConfiguration config) throws Exception {
             handlerId = configId;
         }
 
-        // Setup options
-        final MqttConnectOptions options = setupOptions(config);
+        connectOptions = setupOptions(config);
 
         // Start client (blocking)
         logger.debug("Connecting MQTT client with ID {}", clientId);
         client = new MqttClient(broker, clientId);
         client.setCallback(this);
         client.setManualAcks(true);
-        client.connect(options);
+        try {
+            client.connect(connectOptions);
+        } catch (MqttException e) {
+            if(e.getCause() instanceof ConnectException) {
+                connectionLost(e);
+            } else {
+                throw e;
+            }
+        }
+
+        subscribe();
```

Comment:
We should call subscribe and log the "MQTT client started" trace inside the try block: if the connect fails, we shouldn't try to subscribe